backend/hyper/indexing.go

```diff
@@ -356,6 +356,12 @@ func (bs *indexer) indexChange(idx *indexingCtx, id int64, c cid.Cid, v Change)
 			sb.Meta = title
 		}
 		blocks, ok := v.Patch["blocks"].(map[string]any)
+
+		res, err := hypersql.EntitiesLookupRemovedRecord(idx.conn, sb.Resource.ID.String())
+		if err == nil && res.DeletedResourcesIRI == sb.Resource.ID.String() {
+			return fmt.Errorf("Change belongs to a deleted document [%s]", res.DeletedResourcesIRI)
+		}
```

Comment:
@juligasa Don't we need to do the same for Accounts?

Also, I wonder if this error could break the syncing with a peer that would give you the deleted data. If Alice syncs with Bob, and Bob gives her a bunch of good blobs, along with something that is related to something she has deleted, I believe the entire syncing process would get stuck in failure forever, because this error is not handled during syncing.

Reply:
I just included that in the test and it syncs the OK blobs but not the deleted ones

Reply:
I just included that in the test and it syncs the OK blobs but not the deleted ones.
Adding accounts